2.5 release notes (#540)
ezsystems/developer-documentation: docs/releases/ez_platform_v2.5.md

### Back Office improvements

Several Back Office improvements to facilitate editorial experience, including:

- [Icons for Content Types and the ability to define them](../guide/extending_ez_platform.md#custom-content-type-icons)
- Ability to collapse and expand content preview to have easier access to the Sub-items list
- Responsive Sub-items table with selectable column layout
- Simpler assigning of Object States to content

![Back Office improvements](img/2.5_back_office_improvements.png)

### Matrix Field Type

The new Matrix Field Type enables you to store a table of data.
Columns in the matrix are defined in the Field definition.

![Configuring a Matrix Field Type](img/2.5_matrix_ft.png)

### Product version preview

The Dashboard now shows the version of eZ Platform you are running.

TODO add screenshot with final version

### GraphQL

### API improvements

New API improvements include:

- `sudo()` exposed officially in API to make it more clear how you can skip permission checks when needed
- `AssignSectionToSubtreeSignal` to assign Sections to subtrees
- new `loadLanguageListByCode()` and `loadLanguageListById()` endpoints for bulk loading of languages

> - new method `ContentService->loadContentInfoList()` optimizes logic in `RelationList` and `ParameterProvider`

Reviewer: ```suggestion
- new method `ContentService->loadContentInfoList()` for bulk loading content info
    - Can be used with `ContentService->loadContentListByContentInfo()` to bulk load content
    - 2.5 takes advantage of this in among others `RelationList` and `ParameterProvider`
```

You can now hide and reveal Content items from the Back Office.
Hidden content will be unavailable on the front page regardless of permissions or [Location visibility](../guide/content_management.md#location-visibility).

![Icon for hiding content](img/2.5_hide_content_icon.png)

### Back Office improvements

Several Back Office improvements to facilitate editorial experience, including:

- [Icons for Content Types and the ability to define them](../guide/extending_ez_platform.md#custom-content-type-icons)
- Ability to collapse and expand content preview to have easier access to the Sub-items list
- Responsive Sub-items table with selectable column layout
- Simpler assigning of Object States to content

![Back Office improvements](img/2.5_back_office_improvements.png)

### Matrix Field Type

The new Matrix Field Type enables you to store a table of data.
Columns in the matrix are defined in the Field definition.

![Configuring a Matrix Field Type](img/2.5_matrix_ft.png)

### Product version preview

The Dashboard now shows the version of eZ Platform you are running.

TODO add screenshot with final version

### GraphQL

### API improvements

New API improvements include:

> - `sudo()` exposed to skip permission checks

Reviewer: ```suggestion
- `sudo()` exposed officially in API to make it more clear how you can skip permission checks when needed
```

- `AssignSectionToSubtreeSignal` to assign Sections to subtrees
- new `loadLanguageListByCode()` and `loadLanguageListById()` endpoints for bulk loading of languages

As a Developer I'd like API for bulk loading Content Info Items

## Requirements changes

Due to using Webpack Encore, installing and updating eZ Platform now [requires Node.js and yarn](updating_ez_platform.md#3-update-the-app).

This release also changes support for versions of third-party software:

> - Solr 4 is no longer supported. Use Solr 6 instead.

Reviewer: ```suggestion
- Solr 4 is no longer supported. Use Solr 6 instead (Solr 6.6LTS recommended).
```

- Apache 2.2 is no longer supported. Use Apache 2.4 instead.

> - Varnish 4 is no longer supported. Use Varnish 5.1 or higher.

Reviewer: ```suggestion
- Varnish 4 is no longer supported. Use Varnish 5.1 or higher (6.0LTS recommended)
```

# eZ Platform v2.5

**Version number**: v2.5

**Release date**: March 29, 2019

**Release type**: Long Term Supported

## Notable changes

### Webpack Encore

This release introduces [Webpack Encore](https://symfony.com/doc/3.4/frontend.html#webpack-encore)
as the preferred tool for asset management.

> Assetic is still is use, but it will be deprecated in a future version.

Reviewer 2: ```suggestion
Assetic is still in use, but it will be deprecated in a future version.
```
